### PR TITLE
CI

### PR DESCRIPTION
**Pull request by lyie28**
Changes are as follows:

Two CI jobs:

1) Run ruff formatter

2) Run lavague test command and then execute the outputted file

**Codebase changes:**
1. Add test CLI commandm e.g. lavague -i instructions.yaml -c config.py test
2. Add test boolean to build command- by default always false., but with test command, we call build with test boolean set to true
3. If test boolean is true, we run normal code but call modified make_test_action_engine fn instead of make_action_engine, this uses TestActionEngine which won't do the embedder/LLM query part, it will return a default Selenium Python test code instead of generating code to avoid pointless query for software test
4. It tests telemetry with test boolean set to True so it will raise an exception if one occurs
5. Then we execute generated file

### CODE REVIEW

File: .github/workflows/codebase-test.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+    - name: Set up Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: '3.10'
+
+    - name: Install Ruff
+      run: pip install ruff
+
+    - name: Run Ruff on src/lavague
```
Comment (lyie28) [inline .github/workflows/codebase-test.yaml:34]: also run on examples

File: src/lavague/telemetry.py [hunk 2]
```diff
@@ -7,14 +7,56 @@
...
+                    "test": test,
+                },
             )
+        elif TELEMETRY_VAR == "NONE":
+            pass
     except Exception as e:
-        pass
+        if not test:
+            pass
+        else:
```
Comment (JoFrost) [inline src/lavague/telemetry.py:61]: This lacks detail. Can you try to add the original error code and message?

**Review by JoFrost — CHANGES_REQUESTED** [review]
Looks good to me, except for the comment I left regarding the telemetry.

File: examples/configurations/api/test.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import os
```
Comment (lyie28) [inline examples/configurations/api/test.py:1]: maybe full config
I would but None in api_key
no specifying of max_tokens and temperature

File: src/lavague/action_engine.py [hunk 4]
```diff
@@ -151,3 +151,47 @@ def get_action_streaming(self, query: str, html: str) -> Generator[str, None, No
...
         for text in streaming_response.response_gen:
             yield text
+
+
+class TestActionEngine(BaseActionEngine):
+    """
+    TestActionEngine removes any querying and returns a default code - is used to quickly test software
+
+    Args:
+        llm (`LLMPredictorType`):
```
Comment (lyie28) [inline src/lavague/action_engine.py:161]: change the docstrings

File: src/lavague/telemetry.py [hunk 5]
```diff
@@ -7,14 +7,56 @@
 
 
 def send_telemetry(
-    model_name, code, screenshot, html, instruction, url, origin, success
+    model_name,
+    code,
+    screenshot,
+    html,
+    source_nodes,
```
Comment (lyie28) [inline src/lavague/telemetry.py:14]: source_nodes were removed

File: .github/workflows/codebase-test.yaml (hunk 1 above)
Comment (lyie28) [reply .github/workflows/codebase-test.yaml:34]: Done :)

**Pull request merged**